src/lib.rs

```diff
@@ -124,6 +124,13 @@ impl XorName {
         XOR_NAME_BITS
     }
 
+    /// Returns a copy of `self`, with the `i`-th bit flipped.
+    pub fn with_flipped_bit(&self, index: usize) -> XorName {
+        let &XorName(mut bytes) = self;
+        bytes[index / 8] = bytes[index / 8] ^ (1 << (7 - index % 8));
```

Comment:
>  If malicious code is able to trigger that in our code, I'd still consider our code to be at fault.

Me too.  But we'd potentially still have a running Vault which had failed to respond to a malformed message rather than a crashed Vault.  Having said that, I can't really envisage a scenario right now where we could be "tricked" into giving a bad value to this function as a result of a message.

> Alternatively, we could just define behaviour for other values, e. g. just return an unchanged clone of the name if index >= XOR_NAME_BITS.

If an invalid input is given, I'd prefer a panic to something which looked as though it had worked successfully.

I'm not really convinced by my own argument after discussing it here.  I have a slight preference for this returning `Result<XorName, xor_name::Error>`, but it's maybe overkill indeed, so if no-one else agrees with me, then feel free to just leave as is.